app/src/main/java/com/kickstarter/ui/activities/ChangePasswordActivity.kt

```diff
@@ -40,71 +40,50 @@ class ChangePasswordActivity : AppCompatActivity() {
         this.getEnvironment()?.let { env ->
             viewModelFactory = ChangePasswordViewModel.Factory(env)
         }
-        binding = ActivityChangePasswordBinding.inflate(layoutInflater)
-
-        setContentView(binding.root)
-        setSupportActionBar(binding.changePasswordActivityToolbar.changePasswordToolbar)
-
-        this.logout = getEnvironment()?.logout()
-
-        this.viewModel.outputs.progressBarIsVisible()
-            .compose(Transformers.observeForUIV2())
-            .subscribe {
-                binding.progressBar.isGone = !it
+        setContent {
+            // Replace with feature flag
+            var darkMode by remember { mutableStateOf(false) }
+
+            var showProgressBar by remember { mutableStateOf(false) }
+
+            var errorMessage by remember { mutableStateOf("") }
+
+            KSTheme(useDarkTheme = darkMode) {
+                ChangePasswordScreen(
+                    onBackClicked = { onBackPressedDispatcher.onBackPressed() },
+                    onAcceptButtonClicked = { current, new ->
+                        viewModel.updatePasswordData(current, new)
+                        viewModel.inputs.changePasswordClicked()
+                    },
+                    showProgressBar = showProgressBar,
+                    errorMessage = errorMessage
+                )
             }
-            .addToDisposable(disposables)
 
-        this.viewModel.outputs.passwordWarning()
-            .compose(Transformers.observeForUIV2())
-            .subscribe {
-                binding.warning.text = when {
-                    it == 0 -> null
-                    it != null -> getString(it)
-                    else -> null
+            this.viewModel.outputs.progressBarIsVisible()
```

Comment:
fyi, if you like you could use `subscribeAsState` and work with State, you have an example on `ReportProjectActivity:82`

<img width="741" alt="Screenshot 2023-07-18 at 7 57 17 AM" src="https://github.com/kickstarter/android-oss/assets/4083656/f1119226-6ce9-4173-842d-023d4758906b">

https://developer.android.com/jetpack/compose/state#use-other-types-of-state-in-jetpack-compose

Reply:
oooo, i didnt know about this, will look into it!